### PR TITLE
Few debian updates

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

**Comment by damencho** [comment]
turnserver still does not work on Debian10 because of permissions ... https://github.com/jitsi/jitsi-meet/pull/5714

**Comment by damencho** [comment]
I will push that commit from the other PR here just to test it.

**Comment by damencho** [comment]
Finally I made it work :)

**Comment by damencho** [comment]
@saghul I think this and the meta-package PR are all the known problems so far fixed. I was able to repro several times the re-ordering of the installing packets on Debian10 and with these PRs it is fixed.
Also, I tested with apache, with Nginx already listening on 443 ... and we don't need --no-recommends instructions no more.

**Comment by paulmenzel** [comment]
Commit *debian: Update coturn udp port to non-privileged one.* misses the explanation, why this change is good. The whole point of using the TURN server Coturn is for users behind Firewalls with only ports 80 and 443 open, isn’t it?

**Comment by saghul** [comment]
>  The whole point of using the TURN server Coturn is for users behind Firewalls with only ports 80 and 443 open, isn’t it?

No, it isn't. If 2 participants are behind symmetric NATs they will need TURN to etablish a P2P session.

The reuse of ports 80 / 443 is is indeed nice to try and circumvent spartan firewalls, but the problem here is that Debian 10 doesn't have the capability to bind to priviledged ports set on their unit file, and hacking it ourselves is just not nice.

This is just our default, you are welcome to make changes to your installation of course.

**Comment by paulmenzel** [comment]
I’d prefer if the scripts would create a [drop-in snippet](https://www.freedesktop.org/software/systemd/man/systemd.unit.html) for Coturn.

    $ cat /etc/systemd/system/turnserver.service.d/allow-bind-443.conf
    [Service]
    AmbientCapabilities=CAP_NET_BIND_SERVICE

(`systemctl edit coturn.service` would be for manually creating the drop-in.)

**Comment by saghul** [comment]
We have spent countless hours trying to get this right, we'll just land it as is.

**Comment by paulmenzel** [comment]
I am sorry, that you had to spend so much time with this, and thank you for supporting Jitsi users.

What is the actual problem of sharing the port? The commit message and pull request description do not mention this.

**Comment by julienfastre** [comment]
The issue is here: https://github.com/jitsi/jitsi-meet/issues/5529

TLDR: There are two problems:

* the `turnserver` user does not have any permission to bind on port 443/udp (because this is a privilegied port) ;
* the `turnserver` user does not have sufficient permissions to read the letsencrypt certificates.

File: doc/debian/jitsi-meet-turn/coturn-certbot-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+set -e
+
+COTURN_CERT_DIR="/etc/coturn/certs"
+
+# create a directory to store certs if it does not exists
+if [ ! -d "$COTURN_CERT_DIR" ]; then
+	mkdir -p /etc/coturn/certs
```
Comment (paulmenzel) [inline doc/debian/jitsi-meet-turn/coturn-certbot-deploy.sh:10]: Use the variable `COTURN_CERT_DIR` here?

File: doc/debian/jitsi-meet-turn/coturn-certbot-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	chown -R turnserver:turnserver /etc/coturn/
+	chmod -R 700 /etc/coturn/
+fi
+
+for domain in $RENEWED_DOMAINS; do
+        case $domain in
+        jitsi-meet.example.com)
```
Comment (paulmenzel) [inline doc/debian/jitsi-meet-turn/coturn-certbot-deploy.sh:17]: Add a comment that this is updated/replaced with the correct domain by the install script?

**Comment by paulmenzel** [comment]
> The issue is here: #5529
> 
> TLDR: There are two problems:

Thank you for the summary.

>     * the `turnserver` user does not have any permission to bind on port 443/udp (because this is a privilegied port) ;

Instead of changing to an unprivileged port, I am proposing to extend Coturn’s service unit to give the needed capability to bind to a privileged port to keep “The reuse of ports 80 / 443 is is indeed nice to try and circumvent spartan firewalls,” working.

>     * the `turnserver` user does not have sufficient permissions to read the letsencrypt certificates.

Yes, fixed by commit *add scripts for deploying coturn with certbot*.

**Comment by saghul** [comment]
> What is the actual problem of sharing the port? The commit message and pull request description do not mention this.

The problem is not with sharing, it's with coturn not being able to bind to a priviledged port by default, on Debian 10. This is not the case on Ubuntu for example.

**Comment by damencho** [comment]
Oh, there are more cases here with this turn ... if we skip its configuration, actually it is installed ... we may need to just configure it on another port (change prosody config) ...

**Comment by paulmenzel** [comment]
> Oh, there are more cases here with this turn ...

Do you mean problems? Which are those?

> if we skip its configuration, actually it is installed ...

What do you mean? The package *coturn* was always installed, wasn’t it?

> we may need to just configure it on another port (change prosody config) ...

Why? To fix what problems?

PS: It’s quite hard to help in any way with these terse (short) comments.

**Comment by damencho** [comment]
Yeah, sorry for those. They were more like self note and @saul notes.

Even when we skip configuring coturn it is installed as it is dependency of this package. And we still advertise it.
So what I was thinking last night is to always configuure it and depending on whether there is another nginx site listening on 443 either multiplex or configure coturn and advertise the non 4443 port and for tcp and add instructions the this port may need port forwarding.

**Comment by damencho** [comment]
I was also to leave and coturn and for apache on the non standard port.

**Comment by damencho** [comment]
I did a few more changes. Hope those are the last one. I will be testing those now.

**Comment by damencho** [comment]
Jenkins, skip ci.

**Comment by damencho** [comment]
@paulmenzel I think everything is covered now. If you can take a look. @saghul will also take a look and approve. But I think this is it and this will go to stable fixing all discovered issues with the packaging.

**Comment by Yetangitu** [comment]
Please have a look at my comment in the 'Move UDP port to 4446' thread: https://github.com/jitsi/jitsi-meet/pull/5714#issuecomment-615491997

STUN and TURN have officially designated ports, `3478` and (TLS) `5349`. Some sites already have turn/stun-servers and the accompanying router and firewall configurations, generally set for those official ports. I suggest using these ports instead of 4446/4445.

**Comment by damencho** [comment]
We are discussing this already, to drop the nginx multiplexing and just leave a documentation how to do it. And when this happen we will install and configure turnserver by default even with apache and leave ports on their defaults. We just haven't been able to work on that.

**Comment by RubensRainelli** [comment]
@damencho please fix it ASAP since it broke me 2 servers while reloading NGINX

**Comment by damencho** [comment]
What did broke is more important, as it should not break anything at the moment.

**Comment by RubensRainelli** [comment]
And it will break if I didn't made my own fix xD

**Comment by damencho** [comment]
The current stable detects and if there is already a host listening to 443 the multiplexing is skipped.

**Comment by paulmenzel** [comment]
@RubensRainelli, can you please create an issue with more context and refer to this merge/pull request as the “culprit”?
